avail/src/main/kotlin/avail/anvil/debugger/AvailDebugger.kt

```diff
@@ -283,21 +291,62 @@ class AvailDebugger internal constructor (
 		}
 	}
 
-	/** Renders variables in the variables list view. */
-	class VariablesRenderer : DefaultListCellRenderer()
+	/**
+	 * Renders variables in the top level (excluding the invisible root) of the
+	 * variables tree view.
+	 */
+	class VariableRenderer : DefaultTreeCellRenderer()
 	{
-		override fun getListCellRendererComponent(
-			list: JList<*>?,
-			variable: Any?,
-			index: Int,
-			isSelected: Boolean,
-			cellHasFocus: Boolean
-		): Component = super.getListCellRendererComponent(
-			list,
-			(variable as Variable).presentationString,
-			index,
-			isSelected,
-			cellHasFocus)
+		override fun getTreeCellRendererComponent(
+			tree: JTree,
+			value: Any?,
+			sel: Boolean,
+			expanded: Boolean,
+			leaf: Boolean,
+			row: Int,
+			hasFocus: Boolean
+		): Component
+		{
+			val name = (value as LazyTreeNode).name()
+			return super.getTreeCellRendererComponent(
+				tree, name, sel, expanded, leaf, row, hasFocus)
+		}
+	}
+
+	class LazyTreeNode(
+		val value: Any?
+	) : DefaultMutableTreeNode()
+	{
+		private var name: String? = null
```

Comment:
If you want to avoid a lock here because the computation is idempotent, you should use `LazyThreadSafetyMode.PUBLICATION` (or even `LazyThreadSafetyMode.NONE` in this case, since it's only expanded in the UI thread).